ranch/transit.py

```diff
@@ -803,8 +786,10 @@ def route_bus_link_osmnx_from_start_to_end(
                     shape_polygon = Polygon(zip(lon_list, lat_list))
 
                 # get roadway links and nodes within bounding box
+                # exclude cycleway and footway
                 links_within_polygon_gdf = links_gdf[
-                    links_gdf.geometry.within(shape_polygon)
+                    (links_gdf.geometry.within(shape_polygon)) &
+                    (links_gdf.drive_access == 1)
```

Comment:
@yueshuaing if bus_only exits, then we should also include the bus_only == 1 links, which could be drive_access ==0